core/arch/arm/plat-stm32mp1/drivers/stm32mp1_rcc.c

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (c) 2018, STMicroelectronics
+ */
+
+#include <drivers/stm32mp1_rcc.h>
+#include <io.h>
+#include <mm/core_memprot.h>
+#include <platform_config.h>
+
+uintptr_t stm32_rcc_base(void)
```

Comment:
This pattern is really appearing too many times I think (in this patchset but also more generally in `plat-*/*.c`).
Could `io_pa_or_va()` be used instead? It would have to be moved from `include/drivers/serial.h` to a more generic file, `include/drivers/io.h` maybe?

Reply:
An issue is that some device may be secure or non secure and accessing the device should be done according to its secure state. So we must carefully select the appropriate mapping: non secure when the IP is non secure, and secure when it is secure. The scheme used is the get the device state from the DT (status= / secure-status=) and to setup the device driver accordingly.

Reply:
Ok, I moved to `phys_to_virt_io()` and will see how `io_pa_or_va()` can be moved to a more generic component.